rustls/src/trait_macros.rs

```diff
@@ -0,0 +1,33 @@
+/// pub trait that includes Send & Sync - supports use with alloc::sync::Arc
+#[cfg(not(feature = "withrcalias"))]
+macro_rules! pub_api_trait {
+    ($name:ident, $body:tt) => {
+        pub trait $name: core::fmt::Debug + Send + Sync $body
```

Comment:
I haven't tried this, but I was assuming this would work:

```rust
#[cfg(target_has_atomic = "ptr")]
pub trait ThreadSafe : Send + Sync {}

#[cfg(not(target_has_atomic = "ptr"))]
pub trait ThreadSafe {}
```

(then each extension point trait would be `pub trait ResolvesServerCert: Debug + ThreadSafe { ... }`)

Reply:
Oh, good idea.

Reply:
I did already try this kind of idea, discovered that API user would need to add additional empty `impl` block for the added "ThreadSafe" trait. Looks like Rust unstable would support trait alias, not sure how well this would work for conditional usage of the `Send + Sync` marker traits.